API returned extra features I didn't specify in tfconfig for azurerm_signalr_service

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->
Terraform v0.12.24
+ provider.azurerm (2.18.0)

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_signalr_service`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-ss-test01"
  location = "eastus2"
}

resource "azurerm_signalr_service" "test" {
  name                = "acctestSignalR-test01"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name

  sku {
    name     = "Free_F1"
    capacity = 1
  }

  features {
    flag  = "ServiceMode"
    value = "Serverless"
  }
}
```

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```
=== RUN   TestAccAzureRMSignalRService_serviceMode
=== PAUSE TestAccAzureRMSignalRService_serviceMode
=== CONT  TestAccAzureRMSignalRService_serviceMode
    TestAccAzureRMSignalRService_serviceMode: testing.go:684: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: azurerm_signalr_service.test
          cors.#:                      "1" => "1"
          cors.0.allowed_origins.#:    "1" => "1"
          cors.0.allowed_origins.0:    "*" => "*"
          features.#:                  "3" => "1"
          features.0.flag:             "ServiceMode" => "ServiceMode"
          features.0.value:            "Serverless" => "Serverless"
          features.1.flag:             "EnableConnectivityLogs" => ""
          features.1.value:            "False" => ""
          features.2.flag:             "EnableMessagingLogs" => ""
          features.2.value:            "False" => ""
          hostname:                    "xxx.service.signalr.net" => "xxx.service.signalr.net"
          id:                          "/subscriptions/xx-xx-xx-xx/resourceGroups/acctestRG-200710132141064566/providers/Microsoft.SignalRService/SignalR/xxx" => "/subscriptions/xx-xx-xx-xx/resourceGroups/acctestRG-200710132141064566/providers/Microsoft.SignalRService/SignalR/xxx"
          ip_address:                  "xxx" => "xxx"
          location:                    "eastus2" => "eastus2"
          name:                        "xxx" => "xxx"
          primary_access_key:          "xxx" => "xxx"
          primary_connection_string:   "Endpoint=https://xxx.service.signalr.net;AccessKey=xxx;Version=1.0;" => "Endpoint=https://xxx.service.signalr.net;AccessKey=xxxx;Version=1.0;"
          public_port:                 "443" => "443"
          resource_group_name:         "acctestRG-200710132141064566" => "acctestRG-200710132141064566"
          secondary_access_key:        "xxx" => "xxx"
          secondary_connection_string: "Endpoint=https://xxx.service.signalr.net;AccessKey=xxxx;Version=1.0;" => "Endpoint=https://xxx.service.signalr.net;AccessKey=xxxx;Version=1.0;"
          server_port:                 "443" => "443"
          sku.#:                       "1" => "1"
          sku.0.capacity:              "1" => "1"
          sku.0.name:                  "Free_F1" => "Free_F1"
        
        
        
        STATE:
        
        azurerm_resource_group.test:
          ID = /subscriptions/xx-xx-xx-xx/resourceGroups/acctestRG-200710132141064566
          provider = provider.azurerm
          location = eastus2
          name = acctestRG-200710132141064566
        azurerm_signalr_service.test:
          ID = /subscriptions/xx-xx-xx-xx/resourceGroups/acctestRG-200710132141064566/providers/Microsoft.SignalRService/SignalR/xxx
          provider = provider.azurerm
          cors.# = 1
          cors.0.allowed_origins.# = 1
          cors.0.allowed_origins.0 = *
          features.# = 3
          features.0.flag = ServiceMode
          features.0.value = Serverless
          features.1.flag = EnableConnectivityLogs
          features.1.value = False
          features.2.flag = EnableMessagingLogs
          features.2.value = False
          hostname = xxx.service.signalr.net
          ip_address = xxxx
          location = eastus2
          name = xxx
          primary_access_key = xxxx
          primary_connection_string = Endpoint=https://xxxx.service.signalr.net;AccessKey=xxx;Version=1.0;
          public_port = 443
          resource_group_name = acctestRG-200710132141064566
          secondary_access_key = xxxx
          secondary_connection_string = Endpoint=https://xxxx.service.signalr.net;AccessKey=xxx;Version=1.0;
          server_port = 443
          sku.# = 1
          sku.0.capacity = 1
          sku.0.name = Free_F1
        
          Dependencies:
            azurerm_resource_group.test
--- FAIL: TestAccAzureRMSignalRService_serviceMode (568.47s)
FAIL

Debugger finished with exit code 0

```

### Expected Behavior
It should only returned one feature I specified in tfconfig.
<!--- What should have happened? --->

### Actual Behavior
API returned three features which includes I didn't specified in tfconfig.
<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

--- a/azurerm/internal/services/signalr/resource_arm_signalr_service.go
+++ b/azurerm/internal/services/signalr/resource_arm_signalr_service.go
@@ -21,9 +21,9 @@ import (
 
 func resourceArmSignalRService() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceArmSignalRServiceCreateUpdate,
+		Create: resourceArmSignalRServiceCreate,
 		Read:   resourceArmSignalRServiceRead,
-		Update: resourceArmSignalRServiceCreateUpdate,
+		Update: resourceArmSignalRServiceUpdate,
 		Delete: resourceArmSignalRServiceDelete,
 
 		Timeouts: &schema.ResourceTimeout{
@@ -164,33 +164,31 @@ func resourceArmSignalRService() *schema.Resource {
 	}
 }
 
-func resourceArmSignalRServiceCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+func resourceArmSignalRServiceCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).SignalR.Client
-	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	name := d.Get("name").(string)
 	location := azure.NormalizeLocation(d.Get("location").(string))
 	resourceGroup := d.Get("resource_group_name").(string)
 
+	existing, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("Error checking for presence of existing SignalR %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_signalr_service", *existing.ID)
+	}
+
 	sku := d.Get("sku").([]interface{})
 	t := d.Get("tags").(map[string]interface{})
 	featureFlags := d.Get("features").([]interface{})
 	cors := d.Get("cors").([]interface{})
 	expandedTags := tags.Expand(t)
-
-	if d.IsNewResource() {
-		existing, err := client.Get(ctx, resourceGroup, name)
-		if err != nil {
-			if !utils.ResponseWasNotFound(existing.Response) {
-				return fmt.Errorf("Error checking for presence of existing SignalR %q (Resource Group %q): %+v", name, resourceGroup, err)
-			}
-		}
-
-		if existing.ID != nil && *existing.ID != "" {
-			return tf.ImportAsExistsError("azurerm_signalr_service", *existing.ID)
-		}
-	}
 
 	properties := &signalr.CreateOrUpdateProperties{
 		Cors:     expandSignalRCors(cors),
@@ -217,11 +215,11 @@ func resourceArmSignalRServiceCreateUpdate(d *schema.ResourceData, meta interfac
 		return err
 	}
 	if read.ID == nil {
-		return fmt.Errorf("SignalR %q (Resource Group %q) ID is empty", name, resourceGroup)
+		return fmt.Errorf("SignalR Service %q (Resource Group %q) ID is empty", name, resourceGroup)
 	}
 	d.SetId(*read.ID)
 
-	return resourceArmSignalRServiceRead(d, meta)
+	return resourceArmSignalRServiceUpdate(d, meta)
 }
 
 func resourceArmSignalRServiceRead(d *schema.ResourceData, meta interface{}) error {
@@ -283,6 +281,53 @@ func resourceArmSignalRServiceRead(d *schema.ResourceData, meta interface{}) err
 	return tags.FlattenAndSet(d, resp.Tags)
 }
 
+func resourceArmSignalRServiceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SignalR.Client
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SignalRServiceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	parameters := &signalr.UpdateParameters{}
+
+	if d.HasChanges("cors", "features") {
+		parameters.Properties = &signalr.CreateOrUpdateProperties{}
+
+		if d.HasChange("cors") {
+			corsRaw := d.Get("cors").([]interface{})
+			parameters.Properties.Cors = expandSignalRCors(corsRaw)
+		}
+
+		if d.HasChange("features") {
+			featuresRaw := d.Get("features").([]interface{})
+			parameters.Properties.Features = expandSignalRFeatures(featuresRaw)
+		}
+	}
+
+	if d.HasChange("sku") {
+		sku := d.Get("sku").([]interface{})
+		parameters.Sku = expandSignalRServiceSku(sku)
+	}
+
+	if d.HasChange("tags") {
+		tagsRaw := d.Get("tags").(map[string]interface{})
+		parameters.Tags = tags.Expand(tagsRaw)
+	}
+
+	future, err := client.Update(ctx, id.ResourceGroup, id.Name, parameters)
+	if err != nil {
+		return fmt.Errorf("updating SignalR Service %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for the update of SignalR Service %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	return resourceArmSignalRServiceRead(d, meta)
+}
+
 func resourceArmSignalRServiceDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).SignalR.Client
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
@@ -296,13 +341,13 @@ func resourceArmSignalRServiceDelete(d *schema.ResourceData, meta interface{}) e
 	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)
 	if err != nil {
 		if !response.WasNotFound(future.Response()) {
-			return fmt.Errorf("Error deleting SignalR %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+			return fmt.Errorf("deleting SignalR Service %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 		}
 		return nil
 	}
 	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
 		if !response.WasNotFound(future.Response()) {
-			return fmt.Errorf("Error waiting for the deletion of SignalR %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+			return fmt.Errorf("waiting for the deletion of SignalR Service %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 		}
 	}
 
@@ -326,19 +371,21 @@ func expandSignalRFeatures(input []interface{}) *[]signalr.Feature {
 }
 
 func flattenSignalRFeatures(features *[]signalr.Feature) []interface{} {
-	result := make([]interface{}, 0)
-	if features != nil {
-		for _, feature := range *features {
-			value := ""
-			if feature.Value != nil {
-				value = *feature.Value
-			}
+	if features == nil {
+		return []interface{}{}
+	}
 
-			result = append(result, map[string]interface{}{
-				"flag":  string(feature.Flag),
-				"value": value,
-			})
+	result := make([]interface{}, 0)
+	for _, feature := range *features {
+		value := ""
+		if feature.Value != nil {
+			value = *feature.Value
 		}
+
+		result = append(result, map[string]interface{}{
+			"flag":  string(feature.Flag),
+			"value": value,
+		})
 	}
 	return result
 }
@@ -395,15 +442,20 @@ func flattenSignalRServiceSku(input *signalr.ResourceSku) []interface{} {
 		return []interface{}{}
 	}
 
-	result := make(map[string]interface{})
-
-	if input.Name != nil {
-		result["name"] = *input.Name
-	}
-
+	capacity := 0
 	if input.Capacity != nil {
-		result["capacity"] = *input.Capacity
+		capacity = int(*input.Capacity)
 	}
 
-	return []interface{}{result}
+	name := ""
+	if input.Name != nil {
+		name = *input.Name
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"capacity": capacity,
+			"name":     name,
+		},
+	}
 }

--- a/azurerm/internal/services/signalr/tests/resource_arm_signalr_service_test.go
+++ b/azurerm/internal/services/signalr/tests/resource_arm_signalr_service_test.go
@@ -484,6 +484,16 @@ resource "azurerm_signalr_service" "test" {
   }
 
   features {
+    flag  = "EnableConnectivityLogs"
+    value = "False"
+  }
+
+  features {
+    flag  = "EnableMessagingLogs"
+    value = "False"
+  }
+
+  features {
     flag  = "ServiceMode"
     value = "%s"
   }

--- a/azurerm/internal/services/signalr/tests/resource_arm_signalr_service_test.go
+++ b/azurerm/internal/services/signalr/tests/resource_arm_signalr_service_test.go
@@ -484,6 +484,11 @@ resource "azurerm_signalr_service" "test" {
   }
 
   features {
+    flag  = "ServiceMode"
+    value = "%s"
+  }
+
+  features {
     flag  = "EnableConnectivityLogs"
     value = "False"
   }
@@ -491,11 +496,6 @@ resource "azurerm_signalr_service" "test" {
   features {
     flag  = "EnableMessagingLogs"
     value = "False"
-  }
-
-  features {
-    flag  = "ServiceMode"
-    value = "%s"
   }
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, serviceMode)

